packages/labs/cli/src/test/help_test.ts

```diff
@@ -4,62 +4,232 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {test} from 'uvu';
+import 'source-map-support/register.js';
 // eslint-disable-next-line import/extensions
 import * as assert from 'uvu/assert';
 import {LitCli} from '../lib/lit-cli.js';
 import {LitConsole} from '../lib/console.js';
 import {BufferedWritable} from './buffered-writable.js';
+import {ReferenceToCommand} from '../lib/command.js';
+import {FilesystemTestRig} from './temp-filesystem-rig.js';
+import {ConsoleConstructorOptions} from 'console';
+import * as stream from 'stream';
+import * as pathlib from 'path';
+import {suite} from './uvu-wrapper.js';
 
-let outputStream: BufferedWritable;
-let errorStream: BufferedWritable;
-let cliConsole: LitConsole;
+interface TestContext {
+  console: TestConsole;
+  stdin: NodeJS.ReadableStream;
+  stdinLines: string[];
+  fs: FilesystemTestRig;
+}
 
-test.before(() => {
-  outputStream = new BufferedWritable();
-  errorStream = new BufferedWritable();
-  cliConsole = new LitConsole({
-    stdout: outputStream,
-    stderr: errorStream,
-  });
+const test = suite<TestContext>();
+
+class TestConsole extends LitConsole {
+  readonly outputStream;
+  readonly errorStream;
+  constructor(opts: Partial<ConsoleConstructorOptions> = {}) {
+    const outputStream = new BufferedWritable();
+    const errorStream = new BufferedWritable();
+    super({...opts, stdout: outputStream, stderr: errorStream});
+    this.outputStream = outputStream;
+    this.errorStream = errorStream;
+  }
+}
+
+test.before.each(async (ctx) => {
+  ctx.console = new TestConsole();
+  ctx.fs = new FilesystemTestRig();
+  ctx.stdinLines = [];
+  ctx.stdin = stream.Readable.from(ctx.stdinLines);
```

Comment:
So if I understand how this works, you need to push onto `stdinLines` before the CLI would have done the read from stdin, right?

Reply:
Yes exactly, and generally before you start the CLI running

Reply:
Added a comment.